src/providers/autocomplete/AutocompleteModel.hpp

```diff
@@ -0,0 +1,38 @@
+#pragma once
+
+#include "providers/autocomplete/AutocompleteSource.hpp"
+#include "widgets/listview/GenericListModel.hpp"
+
+#include <QObject>
+#include <QString>
+
+namespace chatterino {
+
+/// @brief Represents the kind of completion occurring
+enum class CompletionKind { Emote, User };
+
+/// @brief AutocompleteModel is a GenericListModel intended to provide completion
+/// suggestions to an InputCompletionPopup. The popup can determine the appropriate
+/// source based on the current input and the user's preferences.
+class AutocompleteModel : public GenericListModel
+{
+public:
+    AutocompleteModel(QObject *parent = nullptr);
```

Comment:
Remove the default parameter, force users to specify nullptr if they really don't want to be the parent to this model

Make the constructor explicit if possible 
```suggestion
    AutocompleteModel(QObject *parent = nullptr);
```